src/Transport/Addressing/Sanitization/Strategies/EndpointOrientedTopologySanitization.cs

```diff
@@ -4,21 +4,13 @@
 
     public class EndpointOrientedTopologySanitization : ISanitizationStrategy
     {
-        IValidationStrategy validationStrategy;
-
-        public EndpointOrientedTopologySanitization(IValidationStrategy validationStrategy)
-        {
-            this.validationStrategy = validationStrategy;
-        }
-
-        public string Sanitize(string entityPathOrName, EntityType entityType)
+        public string Sanitize(string entityPathOrName, EntityType entityType, ValidationResult validationResult)
         {
             // remove invalid characters
             var rgx = new Regex(@"[^a-zA-Z0-9\-\._]");
             entityPathOrName = rgx.Replace(entityPathOrName, "");
 
-            var validationResult = validationStrategy.IsValid(entityPathOrName, entityType);
-            if (!validationResult.IsValid)
+            if (!validationResult.LengthIsValid)
```

Comment:
Correct, that's again highlighting the issue [here](https://github.com/Particular/NServiceBus.AzureServiceBus/issues/252#issuecomment-225677505) :(